velox/docs/functions/presto/decimal.rst

```diff
@@ -159,6 +159,39 @@ then divide by B and round to the nearest whole. This method works as long as
 rescale factor does not exceed 38. If :code:`s + s2 - s1` exceeds 38, an error
 is raised.
 
+For the modulus operation :code:`a % b`, when a and b are integers, the result
+`r` is less than `b` and less than or equal to `a`. Hence the number of digits
+needed to represent `r` is no more than the minimum of the number of digits
+needed to represent `a` or `b`. We can extend this to decimal inputs `a` and
+`b` by computing the modulus of their unscaled values. However, we should
+first make sure that `a` and `b` have the same scale. This can be achieved by
+scaling up the input with lesser scale by the difference in the inputs' scales,
+so both `a` and `b` have scale s. Once `a` and `b` have the same scale, we
+compute the modulus of their unscaled values, A and B. `r` has s digits after
+the decimal point, and since `r` does not need any more digits than the
+minimum number of digits needed to represent `a` or `b`, the result precision
+needs to be increased by the smaller of the differences in the precision and
+scale of either inputs. Hence the result type precision and scale are:
+
+::
+
+	s = max(s1, s2)
+    p = min(p2 - s2, p1 - s1) + max(s1, s2)
+
+Assuming `a` and `b` have unscaled values A' and B' before they are rescaled, we
```

Comment:
I find this paragraph confusing.  If A is unscaled value of 'a' before rescaling, then a = A / 10^s seems wrong. Do we need this paragraph at all? Consider deleting.

Reply:
Thanks for pointing it out, deleted this para.